### PR TITLE
FEATURE : Added load more button for fetching task

### PR DESCRIPTION
**Pull request by sahsisunny**
## Depend on the below PR

- [x]  https://github.com/Real-Dev-Squad/website-backend/pull/1211

### Description:
In this PR we added functionality to load more using intersection observer. When user scroll to the bottom of the page, it will load more data.

### Dev Tested:
- [x] Yes

### Test Coverage
image

### Images/video of the change:

https://github.com/Real-Dev-Squad/website-status/assets/70854507/269586b8-d7ed-4532-ac45-5f5d7db88949

### CODE REVIEW

File: src/styles/tasks.module.scss [hunk 1]
```diff
@@ -33,3 +33,22 @@
...
+
+.loadMoreButton {
+    border: none;
+    border-radius: 0.5rem;
+    background-color: transparent;
+    font-weight: bold;
+    cursor: pointer;
+    transition: all 0.3s ease-in-out;
+
+    &:hover {
```
Comment (vinit717) [inline src/styles/tasks.module.scss:48]: use variable

**Review by Pratiyushkumar — DISMISSED** [review]
LGTM

**Review by vinit717 — APPROVED** [review]

File: __tests__/Unit/Components/Tasks/TasksContent.test.tsx [hunk 2]
```diff
@@ -84,4 +89,60 @@ describe('tasks content', () => {
...
+        const loadMoreButton = await screen.getByRole('button', {
+            name: /load more/i,
+        });
+        expect(loadMoreButton).toBeEnabled();
+        fireEvent.click(loadMoreButton);
+        const task2 = await findByText(
+            'Design and develop an online booking system'
+        );
+        expect(task2).toBeInTheDocument();
+    });
```
Comment (bharati-21) [inline __tests__/Unit/Components/Tasks/TasksContent.test.tsx:147]: Can we write a test to check if one of the tabs is clicked, then the tasks are only related to that satus?

File: src/components/tasks/TasksContent.tsx [hunk 3]
```diff
@@ -57,44 +28,95 @@ export const TasksContent = () => {
...
-        if (isFetching) {
-            return <TasksLoader />;
-        }
+    const tasksNotAvailable =
+        tasks === undefined || tasksGroupedByStatus[tab] === undefined;
 
-        if (dev === 'true' && tasksData.tasks?.length) {
-            return <TaskList tasks={tasksData.tasks} />;
-        }
+    if (tasksNotAvailable || tasks.length === 0) {
```
Comment (bharati-21) [inline src/components/tasks/TasksContent.tsx:34]: why not add `!tasks.length` in the constant `tasksNotAvailable`?

**Review by vinit717 — APPROVED** [review]

**Review by fakhruddinkw — APPROVED** [review]

File: src/app/services/tasksApi.ts [hunk 4]
```diff
@@ -5,23 +5,31 @@ import task, {
...
-                    url = nextPage;
+                if (nextTasks) {
+                    url = nextTasks;
                 }
 
-                if (prevPage) {
-                    url = prevPage;
+                if (prevTasks) {
+                    url = prevTasks;
                 }
```
Comment (ankushdharkar) [inline src/app/services/tasksApi.ts:32]: Let's rewrite this everyone.

**Pull request merged**